src/components/markdown-renderer/hooks/use-extract-first-headline.ts

```diff
@@ -0,0 +1,24 @@
+import React, { useCallback, useEffect } from 'react'
+
+export const useExtractFirstHeadline = (documentElement: React.RefObject<HTMLDivElement>, onFirstHeadingChange?: (firstHeading: string | undefined) => void) => {
+  const extractInnerText = useCallback((node: ChildNode): string => {
+    let innerText = ''
+    if (node.childNodes && node.childNodes.length > 0) {
+      node.childNodes.forEach((child) => { innerText += extractInnerText(child) })
+    } else if (node.nodeName === 'IMG') {
+      innerText += (node as HTMLImageElement).getAttribute('alt')
+    } else {
+      innerText += node.textContent
+    }
+    return innerText
+  }, [])
+
+  useEffect(() => {
+    if (onFirstHeadingChange && documentElement.current) {
+      const firstHeading = documentElement.current.getElementsByTagName('h1').item(0)
+      if (firstHeading) {
+        onFirstHeadingChange(extractInnerText(firstHeading))
+      }
+    }
+  }, [documentElement, extractInnerText, onFirstHeadingChange])
```

Comment:
this needs to be given the content and the documentElement as seen here
https://github.com/codimd/react-client/blob/b094fe1de66cb0124842c667a5cb9bc27b95e844/src/components/markdown-renderer/full-markdown-renderer.tsx#L133-L140
This code isn't called when the markdown content changes and therefore only extracts the first title when the pad get's opened, hence the cypress errors

Reply:
fixed